File: libgomp/testsuite/libgomp.c/cancel-sections-1.c
/* { dg-do run } */
/* { dg-set-target-env-var OMP_CANCELLATION "true" } */

#include <stdlib.h>
#include <omp.h>

int
main ()
{
  if (!omp_get_cancellation ())
    return 0;
  #pragma omp parallel num_threads (32)
  {
    #pragma omp sections
      {
	{
	  #pragma omp cancel sections
	  abort ();
	}
      #pragma omp section
	{
	  #pragma omp cancel sections
	  abort ();
	}
      #pragma omp section
	{
	  #pragma omp cancel sections
	  abort ();
	}
      #pragma omp section
	{
	  #pragma omp cancel sections
	  abort ();
	}
      }
  }
  return 0;
}

File: libgomp/testsuite/libgomp.fortran/cancel-sections-1.f90
! { dg-do run }
! { dg-set-target-env-var OMP_CANCELLATION "true" }

  use omp_lib

  if (omp_get_cancellation ()) then
    !$omp parallel num_threads(32)
      !$omp sections
	  !$omp cancel sections
	  stop 1
	!$omp section
	  !$omp cancel sections
	  stop 2
	!$omp section
	  !$omp cancel sections
	  stop 3
	!$omp section
	  !$omp cancel sections
	  stop 4
      !$omp end sections
    !$omp end parallel
  end if
end
